tools/functional-tester/etcd-tester/stresser.go

```diff
@@ -99,7 +100,7 @@ func (s *stresser) run(ctx context.Context, kvc pb.KVClient) {
 	defer s.wg.Done()
 
 	for {
-		if err := s.rateLimiter.Wait(ctx); err == context.Canceled {
+		if err := s.rateLimiter.Wait(ctx); err == context.Canceled || ctx.Err() == context.Canceled {
```

Comment:
then wouldn't it be racey regardless? `Wait` already checks `ctx.Done()` before triggering the timer, so if it's returning nil, the timer trigger and the cancel must be concurrent. If the ctx cancel() is concurrent with `<-t.C`, then wouldn't it be possible for the timer to expire and `ctx.Err()` evaluates to `nil` on this check but before the `Put()`, the context is canceled?